satpy/composites/__init__.py

```diff
@@ -1701,3 +1702,85 @@ def __call__(self, projectables, nonprojectables=None, **info):
 
         masked_projectable = projectable.where(lon_min_max)
         return super().__call__([masked_projectable], **info)
+
+
+class MissingTime(Exception):
+    """Raised when temporal composite building lacks time dimension."""
+
+
+class BaseTemporalCompositor(CompositeBase):
+    """Compositors combining multiple time steps.
+
+    Base class for compositors that combine inputs from two or more time steps.
+
+    To use this, the user must start with a :class:`MultiScene` and load the
+    temporal composite from there.  Composite generation will fail due to missing
+    temporal information in the containing scenes, but it will add the temporal
+    composite to the scene wishlists.  Now we run ``MultiScene.blend(timeseries)``,
+    which will create DataArrays with a time dimension.  After blending, the
+    blend method tries again to generate the composites, and on this second run
+    the generation should work.
+    """
+
+    def __init__(self, name, prerequisites, optional_prerequisites=None, **kwargs):
+        """Initialise a temporal compositor."""
+        self._ensure_prerequisites_have_times(prerequisites)
+        super().__init__(name, prerequisites, optional_prerequisites, **kwargs)
+
+    def _check_time_dimension(self, projectables):
+        """Make sure all projectables have a time dimension."""
+        for projectable in projectables:
+            if "time" not in projectable.dims:
+                raise MissingTime(
+                    "Creating temporal composite needs time dimensions. "
+                    "Typically, this comes from starting with a MultiScene "
+                    "and then performing timeseries blending.")
+
+    def _ensure_prerequisites_have_times(self, prerequisites):
+        """Make sure all prerequisites have times defined."""
+        for preq in prerequisites:
+            if "time" not in preq.to_dict():
+                raise KeyError(
+                    "In a temporal composite, all prerequisites should have "
+                    "time information defined.  Time information is missing for "
+                    f"{preq!s}")
+
+    def _apply_temporal_prerequisites(self, projectables):
+        """Apply temporal prerequisites.
+
+        The generic Satpy composite loading logic does not understand time
+        dependencies.  When :meth:`MultiScene.blend` reruns the composite
+        loading for a composite combining the same channel for three time
+        steps, we will be passed the same 3-D data array three times.
+
+        This method matches the temporal prerequisites as defined in the
+        compositor with the time dimension associated with the dataarrays.
+        """
+        # reference time is the newest time
+        ref_time = projectables[0]["time"][-1]
+        new_projectables = []
+        for (dq, proj) in zip(self.attrs["prerequisites"], projectables):
+            new_da = proj.sel(time=ref_time+pd.Timedelta(dq["time"]))
```

Comment:
```suggestion
            new_da = proj.sel(time=ref_time+pd.Timedelta(dq["time"]), method='nearest')
```
This resolves the issue I saw with SEVIRI data.

Reply:
That was much easier than I thought it might be :)